### PR TITLE
[SAL-5204] add submenu for form element

### PR DESCRIPTION
**Pull request by ghost**
Add submenu - three dots in top right corner

[Look](https://www.loom.com/share/7fd717b498cb467ea29ede5511e3a578)

### CODE REVIEW

File: src/utils/hooks/useToggle.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+function useToggle(defaultValue = false) {
+  const [isOpen, setIsOpen] = React.useState(defaultValue);
+
+  const toggle = () => setIsOpen(!isOpen);
+  const close = () => setIsOpen(false);
+
+  return { isOpen, toggle, close };
+}
+
+export default useToggle;
```
Comment (tarasbashuk) [inline src/utils/hooks/useToggle.jsx:12]: ❤️

File: src/utils/hooks/useToggle.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+function useToggle(defaultValue = false) {
+  const [isOpen, setIsOpen] = React.useState(defaultValue);
+
+  const toggle = () => setIsOpen(!isOpen);
```
Comment (tarasbashuk) [inline src/utils/hooks/useToggle.jsx:6]: Here it's not fundamentally, but will be better next one:

```const toggle = () => setIsOpen(isOpen => !isOpen);```

File: src/form-elements/header-bar.jsx [hunk 3]
```diff
@@ -3,16 +3,49 @@
...
+};
 
+export default class HeaderBar extends React.Component {
   render() {
     return (
       <div className="toolbar-header">
         <span className="badge badge-secondary">{this.props.data.text}</span>
-        <div className="toolbar-header-buttons">
+        {this.props.showInlineEditForm && <HeaderMenu />}
+        {!this.props.showInlineEditForm && <div className="toolbar-header-buttons">
```
Comment (tarasbashuk) [inline src/form-elements/header-bar.jsx:48]: You can use ternary operator here

File: src/form-elements/component-header.jsx [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+const ComponentHeader = (props) => {
+  if (props.mutable) {
+    return null;
+  }
+  return (
+    <div>
+      {props.data.pageBreakBefore &&
+        <div className="preview-page-break">Page Break</div>
+      }
+      <HeaderBar parent={props.parent} editModeOn={props.editModeOn} data={props.data} index={props.index} setAsChild={props.setAsChild} onDestroy={props._onDestroy} onEdit={props.onEdit} static={props.data.static} required={props.data.required} showInlineEditForm={props.showInlineEditForm}/>
```
Comment (tarasbashuk) [inline src/form-elements/component-header.jsx:13]: It's very hard to read ))
Maybe every prop from the new line?

File: src/form-elements/header-bar.jsx [hunk 5]
```diff
@@ -3,16 +3,49 @@
...
+const HeaderMenu = () => {
+  const { isOpen, toggle } = useToggle();
+
+  const buttonList = [
+    {
+      icon: 'fas fa-arrows-alt',
+      text: 'Move',
+    },
+    {
+      icon: 'far fa-trash-alt',
```
Comment (tarasbashuk) [inline src/form-elements/header-bar.jsx:18]: It's a kind of fundamental thing:
If we diced to deeply reworked RFB2, maybe could bring "our" icons? 
The same for buttons, inputs, etc.
Or we still want to add our custom components on our side and just tune some staff in this lib?

**Review by david-wolgemuth — APPROVED** [review]
nice 👍

**Pull request merged**